pkg/common/versions/common/utils.go

```diff
@@ -0,0 +1,94 @@
+package common
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver"
+	"github.com/openshift/osde2e/pkg/common/spi"
+)
+
+// NextReleaseAfterGivenVersionFromVersionList will attempt to look for the next valid X.Y stream release, given a delta (releaseFromGivenVersion)
+// Example In/Out
+// In: 4.3.12, [4.3.13, 4.4.0, 4.5.0], 2
+// Out: 4.5.0, nil
+func NextReleaseAfterGivenVersionFromVersionList(givenVersion *semver.Version, versionList []*spi.Version, releasesFromGivenVersion int) (*semver.Version, error) {
```

Comment:
I'm actually not certain. This function is a copy-paste, but I'd be hesitant to do a delta like this -- part of the reason here is that we're not guaranteed that the "next release" is x.y.z+1. If you look at prod, the current enabled releases are 4.3.0, 4.3.12, 4.3.18, and 4.3.19 -- this logic will catch all of those in order rather than literally 0.0.1 more.

Reply:
I know how/why we got to present state, I'm just curious if having the ability to specify a semver-delta rather than a list-delta makes more sense.

For example, if we want to test the next Y-stream nightly we could do 0.1.0-nightly. The next Z-stream would be 0.0.1. It provides us more flexibility and specificity.

That said, maybe the specificity is a moot point when considering the ClusterImageSets...

Reply:
I see what you're saying. This is possible, but I don't know that this would add a whole lot of value. I think I'd want to avoid adding it into this already complicated PR, at any rate. That okay?

Reply:
Yeah, I'm good keeping the PR as-is.